crates/bevy_text/src/font_atlas_set.rs

```diff
@@ -94,43 +96,61 @@ impl FontAtlasSet {
             if !font_atlases.last_mut().unwrap().add_glyph(
                 textures,
                 texture_atlases,
-                glyph_id,
-                glyph_position.into(),
+                layout_glyph.cache_key,
                 &glyph_texture,
+                left,
+                top,
+                w,
+                h,
             ) {
-                return Err(TextError::FailedToAddGlyph(glyph_id));
+                return Err(TextError::FailedToAddGlyph(layout_glyph.cache_key.glyph_id));
             }
         }
 
-        Ok(self
-            .get_glyph_atlas_info(font_size, glyph_id, glyph_position)
-            .unwrap())
+        Ok(self.get_glyph_atlas_info(layout_glyph.cache_key).unwrap())
     }
 
     pub fn get_glyph_atlas_info(
         &mut self,
-        font_size: f32,
-        glyph_id: GlyphId,
-        position: Point,
+        cache_key: cosmic_text::CacheKey,
     ) -> Option<GlyphAtlasInfo> {
         self.font_atlases
-            .get(&FloatOrd(font_size))
+            .get(&cache_key.font_size_bits)
             .and_then(|font_atlases| {
                 font_atlases
                     .iter()
                     .find_map(|atlas| {
                         atlas
-                            .get_glyph_index(glyph_id, position.into())
+                            .get_glyph_index(cache_key)
                             .map(|glyph_index| (glyph_index, atlas.texture_atlas.clone_weak()))
                     })
-                    .map(|(glyph_index, texture_atlas)| GlyphAtlasInfo {
-                        texture_atlas,
-                        glyph_index,
-                    })
+                    .map(
+                        |((glyph_index, left, top, w, h), texture_atlas)| GlyphAtlasInfo {
+                            texture_atlas,
+                            glyph_index,
+                            left,
+                            top,
+                            width: w,
+                            height: h,
+                        },
+                    )
             })
     }
 
     pub fn num_font_atlases(&self) -> usize {
         self.font_atlases.len()
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct PositionedGlyph {
+    pub position: Vec2,
+    pub size: Vec2,
+    pub atlas_info: GlyphAtlasInfo,
```

Comment:
Doc strings for the other fields please <3 I don't have great suggestions, as I'm still trying to understand this code, but feel free to ping me for wording help.